src/charm.py

```diff
@@ -165,6 +167,11 @@ def _on_leader_elected(self, _) -> None:
                 self.set_secret(
                     "app", required_password, generate_random_password(PASSWORD_LENGTH)
                 )
+        self.unit_peer_data.update({"leader": "true"})
+
+    def _on_leader_settings_changed(self, _) -> None:
```

Comment:
is leader_settings_changed emitted for the leader unit after leader_elected is emitted?

Reply:
just for non-leaders, as observed and in docs